File: common/src/main/java/com/pluxity/facility/dto/FacilityPathResponse.java
package com.pluxity.facility.dto;

import com.pluxity.facility.path.FacilityPath;

public record FacilityPathResponse(Long id, String name, String type, String path) {
    public static FacilityPathResponse from(FacilityPath facilityPath) {
        return new FacilityPathResponse(
                facilityPath.getId(),
                facilityPath.getName(),
                facilityPath.getPathType().name(),
                facilityPath.getPath());
    }
}

File: common/src/main/kotlin/com/pluxity/facility/dto/FacilityPathResponse.kt
package com.pluxity.facility.dto

import com.pluxity.facility.path.FacilityPath

data class FacilityPathResponse(
    val id: Long,
    val name: String,
    val type: String,
    val path: String,
)

fun FacilityPath.toPathResponse(): FacilityPathResponse =
    FacilityPathResponse(
        id = this.requiredId(),
        name = this.name,
        type = this.pathType.name,
        path = this.path,
    )
